core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocateUnassignedDecision.java

```diff
@@ -364,6 +364,22 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(totalDelayInMillis);
     }
 
+    /**
+     * A toXContent implementation that leaves off some of the non-critical fields, and assumes the outer object
+     * is created outside of this method call.
+     */
+    public static XContentBuilder discoveryNodeToXContent(XContentBuilder builder, Params params, DiscoveryNode node) throws IOException {
```

Comment:
super nitty, but DiscoveryNode should be the first parameter of this method, so it reads more nicely.